controller/controlplane_extensions/controller_enqueue_funcs.go

```diff
@@ -283,8 +283,23 @@ func enqueueControlPlaneForDataPlane(
 				}
```

Comment:
Some additional comments for the `for controlplane := range controlplanes.Items` loop (the part of code not touched in this PR):
- The usage of the loop looks like to find the `controlPlane` that manages the input `DataPlane`. Do you think using `lo.Find(controlplanes.Items, func(cp ControlPlane) bool {...})` to find the satisfying `ControlPlane` could make the code easier to read?
- The `continue` in the inner loop `for _, ref := range controlplane.OwnerReferences` under `if owner != nil` just jumps out of the iteration of the inner loop but node out of the outer loop. Also, the `ref` defined in the loop is not used elsewhere. Should it jump out of the outer loop?

Reply:
It might refactored as you're proposing. We should just keep in mind when working on this to not overuse the closures (like it's done when utilizing `lo` functions).

Let's do this in a separate PR though.